Documentation describing how to build the custom envs (#259)
gladiaio/gladia: src/venv-builder/README.md

# Custom environments

## How do custom environments are built

> Each task/model can have an `env.yaml` file that describe a custom environment to use.

Reviewer: ```suggestion
Each task/model can have an `env.yaml` file describing the custom environment to use.
```

Notes :

> - If both the task and the model has an `env.yaml` file, the model custom environment will prime fot his specific model.\

Reviewer: ```suggestion
- If both the task and the model have an `env.yaml` file, the model custom environment will take precedence for its own specific model.\
```

> - If a model does not have an `env.yaml` file but the task do, the model will be launched using the `env.yaml` file.

Reviewer: ```suggestion
- If a model does not have an `env.yaml` file but the task do, the model will be launched using the task's `env.yaml` file.
```

An `env.yaml` file is described as follow:
```yaml
inherit:
  - templates-to-inherit-from

dependencies:
  - conda-forge-dependencies
  - pip:
    - pip-dependencies
```

A custom environment can inherit from templates (see [this folder](src/venv-builder/envs)).\

> Templates contains multiple package related to either a modality (i.e image) or a package (i.e pytorch). Templates are useful to build custom environment without having to list each packages each time.

Reviewer: ```suggestion
Templates contain multiple package related to either a modality (i.e image) or a package (i.e pytorch). Templates are useful to build custom environment without having to list each package every time.
```

Note : a custom environment must have either a conda-forge dependency, a pip dependency or a inherit value.

## Usage

> Before building any custom environment be sure to be in the `venv-builder` folder.

Reviewer: ```suggestion
Before building any custom environment, be sure to be in the `venv-builder` folder.
```

> usages:

Reviewer: ```suggestion
Usages:
```
